pkg/container/qemu_runner.go

```diff
@@ -1756,9 +1742,21 @@ func generateCpio(ctx context.Context, cfg *Config) (string, error) {
 
 	cacheDir = filepath.Join(cacheDir, "melange-cpio")
 
+	// Include additional packages in cache filename to invalidate cache when they change
+	additionalPkgs := os.Getenv("QEMU_ADDITIONAL_PACKAGES")
+	cacheSuffix := ""
+	if additionalPkgs != "" {
+		// Use sanitized package list as cache key (replace commas and dots with dashes)
+		sanitized := strings.NewReplacer(",", "-", ".", "-").Replace(additionalPkgs)
+		if len(sanitized) > 32 {
+			sanitized = sanitized[:32] // Limit length for reasonable filenames
```

Comment:
Can we foresee hitting the edge case where we install enough packages to make this consistent even though we're installing new/different packages after the cutoff?

Would something like a checksum work better?